src/ShopBundle/objects/db/TShopCategory.class.php

```diff
@@ -648,8 +648,17 @@ public function getTargetPage()
             if (null === $activePortal) {
                 return null;
             }
-            $targetPageId = $activePortal->GetSystemPageId('products');
-            $defaultPage = TdbCmsTplPage::GetNewInstance($targetPageId);
+
+            $systemPageService = $this->getSystemPageService();
+            $defaultSystemPage = $systemPageService->getSystemPage('products', $activePortal);
```

Comment:
Use a named constant for "products"?

Reply:
Done. It is used at three places in the file now.